### PR TITLE
ABI Encoding tests.

### PR DESCRIPTION
**Pull request by ebadiere**
Test covering ABI Encoding.

**Related issue(s)**: #432 

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: contracts/solidity/encoding/Receiver.sol [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+contract Receiver {
+    event ReceivedData(uint256 data);
+
+    function receiveData(uint256 data) external {
+        emit ReceivedData(data);
+    }
+}
```
Comment (AlfredoG87) [inline contracts/solidity/encoding/Receiver.sol:10]: missing blank line

File: contracts/solidity/encoding/Sender.sol [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        bytes memory payload = abi.encodeCall(
+            Receiver(address(receiver)).receiveData,
+            (data)
+        );
+
+        (bool success,) = address(receiver).call(payload);
+        require(success, "External call using abi.encodeCall failed"); 
+    }
+
+}
```
Comment (AlfredoG87) [inline contracts/solidity/encoding/Sender.sol:30]: missing blank line

**Review by AlfredoG87 — DISMISSED** [review]
LGTM

File: contracts/solidity/encoding/Receiver.sol (hunk 1 above)
Comment (ebadiere) [reply contracts/solidity/encoding/Receiver.sol]: Done.

File: contracts/solidity/encoding/Sender.sol (hunk 2 above)
Comment (ebadiere) [reply contracts/solidity/encoding/Sender.sol]: Done.

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Pull request merged**